packages/backend/src/engine/backend-secret/backend_secret.ts

```diff
@@ -16,9 +16,11 @@ export class CfnTokenBackendSecret implements BackendSecret {
    * The name of the secret to fetch.
    */
   constructor(
-    private readonly name: string,
+    private readonly secretName: string,
     private readonly secretResourceFactory: BackendSecretFetcherFactory
-  ) {}
+  ) {
+    BackendSecretFetcherFactory.registerSecret(secretName);
+  }
```

Comment:
This doesn't look like a healthy data flow.

Could we move this detail into factory (since we do receive reference to factory in this ctor already)?

For example.
Change scope of these variables https://github.com/aws-amplify/amplify-backend/blob/8dd7286bf1a870e69766963e872fad4add68bc84/packages/backend/src/secret.ts#L30-L33 and promote them to global scope (to make factory a singleton)
Use cdk.Lazy to accumulate secret names inside factory?

Reply:
These are singleton factories, except they do not use DI system like other factories, they use the scope.node.tryFindChild('<token>')

Reply:
can `registerSecret` become internal detail of the factory then ?

Reply:
@awsluja this is what I have in mind https://github.com/aws-amplify/amplify-backend/commit/9816f1714dfdfe0d72881d3a4de5a9f5c4999f54 .